fairseq/models/roberta/model.py

```diff
@@ -307,7 +310,13 @@ def forward(self, src_tokens, features_only=False, return_all_hiddens=False, mas
                   is a list of hidden states. Note that the hidden
                   states have shape `(src_len, batch, vocab)`.
         """
-        x, extra = self.extract_features(src_tokens, return_all_hiddens=return_all_hiddens)
+        with torch.set_grad_enabled(not freeze_encoder):
+            was_training = self.training
+            if freeze_encoder:
+                self.eval()  # disable dropout when encoder is frozen
+            x, extra = self.extract_features(src_tokens, return_all_hiddens=return_all_hiddens)
+            if was_training and freeze_encoder:
```

Comment:
@myleott I added this check to make sure we are only enabling training if it has been enabled before